geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -4428,6 +4431,35 @@ public Object getFromLocalBucket(int bucketId, final Object key, final Object aC
     return null;
   }
 
+  boolean triggerWriter(RegionEventImpl event) {
+    CacheWriter localWriter = basicGetWriter();
+    Set netWriteRecipients = localWriter == null ? this.distAdvisor.adviseNetWrite() : null;
+
+    if (localWriter == null && (netWriteRecipients == null || netWriteRecipients.isEmpty())) {
+      return false;
+    }
+
+    final long start = getCachePerfStats().startCacheWriterCall();
+    try {
+      SearchLoadAndWriteProcessor processor = SearchLoadAndWriteProcessor.getProcessor();
+      String theKey = "preDestroyRegion";
+      int paction = 0;
+      if (event.getOperation().isRegionDestroy()) {
+        theKey = "preDestroyRegion";
+        paction = SearchLoadAndWriteProcessor.BEFOREREGIONDESTROY;
```

Comment:
Can the caller pass in the Processor. We could avoid checking for operation.
Also is paction -> processorAction ? can we make it clear. Sorry for nitpicking. Just wanted to be clear with what we are doing.

Reply:
fixed and I removed all the listener part tests